packages/addons/remote-functions/index.ts

```diff
@@ -0,0 +1,210 @@
+import {
+	colors,
+	defineAddon,
+	defineAddonOptions,
+	isVersionUnsupportedBelow
+} from '@sveltejs/cli-core';
+import { common, exports, object, type AstTypes } from '@sveltejs/cli-core/js';
+import { parseScript } from '@sveltejs/cli-core/parsers';
+import { addToDemoPage } from '../common.ts';
+import { capitalCase, kebabCase, pascalCase } from 'change-case';
+
+const defaultDemoName = 'myPost';
+const options = defineAddonOptions({
+	withDemo: {
+		question: `Do you want to include a demo? ${colors.dim('(includes a query & demo page)')}`,
+		type: 'boolean',
+		default: true
+	},
+	demo: {
+		question: 'Which name would you like to use for the demo?',
+		type: 'string',
+		default: '',
+		placeholder: defaultDemoName,
+		condition: ({ withDemo }) => withDemo === true
+	}
+});
+
+export default defineAddon({
+	id: 'remote-functions',
+	shortDescription: 'remote functions',
+	homepage: 'https://svelte.dev/docs/kit/remote-functions',
+	options,
+	setup: ({ kit, unsupported }) => {
+		if (!kit) unsupported('Requires SvelteKit');
+	},
+
+	run: ({ sv, options, dependencyVersion, kit, typescript }) => {
+		if (!kit) throw new Error('SvelteKit is required');
+		const ext = typescript ? 'ts' : 'js';
+
+		// ensure svelte & sveltekit versions
+		const ensureVersion = (pkg: string, atLeast: string, setTo: string) => {
+			const depVersion = dependencyVersion(pkg) ?? '0.0.0';
+			const unsupported = isVersionUnsupportedBelow(depVersion, atLeast);
+			if (unsupported) sv.devDependency(pkg, setTo);
+		};
+		ensureVersion('@sveltejs/kit', '2.27', '^2.27.0');
+		ensureVersion('svelte', '5.36', '^5.36.0');
+
+		// experimental flags
+		sv.file('svelte.config.js', (content) => {
+			const { ast, generateCode } = parseScript(content);
+
+			const { value: config } = exports.createDefault(ast, { fallback: object.create({}) });
+
+			// Handle kit.experimental.remoteFunctions
+			const kitConfig = config.properties.find(
+				(p) => p.type === 'Property' && p.key.type === 'Identifier' && p.key.name === 'kit'
```

Comment:
for this you can probably use `object.property`:

```ts
const kitConfig = object.property(config, {
  name: 'kit',
  fallback: object.create({
    experimental: {
      remoteFunctions: true
    }
  }
});
```

you might be able to do this for each of the nested objects to simplify away the `if`/`else` chains

though you'll still want to check if the `kitConfig.type === 'ObjectExpression'` and probably overwrite it if its not